tools/src/main/scala/scala/scalanative/linker/LinktimeValueResolver.scala

```diff
@@ -9,14 +9,15 @@ trait LinktimeValueResolver { self: Reach =>
 
   private lazy val linktimeProperties = {
     val conf = config.compilerConfig
+    val triple = conf.configuredOrDetectedTriple
     val linktimeInfo = "scala.scalanative.meta.linktimeinfo"
     val predefined: NativeConfig.LinktimeProperites = Map(
       s"$linktimeInfo.debugMode" -> (conf.mode == Mode.debug),
       s"$linktimeInfo.releaseMode" -> (conf.mode == Mode.releaseFast || conf.mode == Mode.releaseFull || conf.mode == Mode.releaseSize),
-      s"$linktimeInfo.isWindows" -> Platform.isWindows,
-      s"$linktimeInfo.isLinux" -> Platform.isLinux,
-      s"$linktimeInfo.isMac" -> Platform.isMac,
-      s"$linktimeInfo.isFreeBSD" -> Platform.isFreeBSD,
+      s"$linktimeInfo.isWindows" -> (triple.os == "windows"),
+      s"$linktimeInfo.isLinux" -> (triple.os == "linux"),
+      s"$linktimeInfo.isMac" -> (triple.vendor == "apple" && triple.os == "darwin"),
+      s"$linktimeInfo.isFreeBSD" -> (triple.os == "freebsd"),
```

Comment:
Actually I was hoping that we can remove these completely and instead define them in the code. e.g. replace
```scala
  @resolvedAtLinktime("scala.scalanative.meta.linktimeinfo.isLinux")
  def isLinux: Boolean = resolved
```
with
```scala
  def isLinux: Boolean = target.os == "linux"
```
but unfortunately the `getLinktimeCondition` does not recognize that properly. I hope we can find a way to fix it, because otherwise it is impossible for users to define their own helper function for checking triples, they must always inline at use-site.

Reply:
I was not aware of that issue, we should be able to workaround it by combining both `@alwaysinline` annotation and making rhs and if-else
This snippet would link if os is not set to `windows`
```
import scala.scalanative.unsafe._
import scala.scalanative.annotation._

@extern
object bindings {
  def nonExistingMethod(): Unit = extern
}
object check {
  @alwaysinline def isWindows =
    if (scala.scalanative.meta.LinktimeInfo.os == "windows") true
    else false
}
object Test {

  def main(args: Array[String]): Unit = {
    println("Hello, World!")
    // if( scala.scalanative.meta.LinktimeInfo.os == "windows")
    if (check.isWindows)
      bindings.nonExistingMethod()
    else println("Ok!")

  }
}
```

I'll try to fix it in the compiler plugin, so we would not require a boilerplate if-else

Reply:
Thanks, I tried some tricks like that, but was having problems here:
https://github.com/scala-native/scala-native/blob/9a8a782f22486136b2ab9c3fa893f9d200182b77/test-interface/src/main/scala/scala/scalanative/testinterface/signalhandling/SignalConfig.scala#L152

Reply:
I have an idea of how we might fix it, I will try it tomorrow. Anyway we should be able to resolve it in follow up PR